tests/EncryptedFS.concurrent.test.ts

```diff
@@ -57,6 +74,42 @@ describe(`${EncryptedFS.name} Concurrency`, () => {
     });
   });
   describe('concurrent inode creation', () => {
+    test('EncryptedFS.open, EncryptedFS.mknod and EncryptedFS.mkdir', async () => {
+      const path1 = pathNode.join('dir', 'file1');
+
+      await fc.assert(
+        fc
+          .asyncProperty(fc.scheduler(), async (s) => {
+            await efs.mkdir('dir');
+            const gate = gateFactory(s);
+            const prom = Promise.allSettled([
+              gate(() => efs.mknod(path1, constants.S_IFREG, 0, 0)),
+              gate(() => efs.mknod(path1, constants.S_IFREG, 0, 0)),
+              gate(() => efs.open(path1, constants.O_RDWR | constants.O_CREAT)),
+              gate(() => efs.open(path1, constants.O_RDWR | constants.O_CREAT)),
+              gate(() => efs.mkdir(path1)),
+              gate(() => efs.mkdir(path1)),
+            ]);
+            await s.waitAll();
+            const results = await prom;
+            results.map((item) => {
+              if (item.status !== 'fulfilled') {
+                // Should fail as a normal FS error
+                expectReason(item, ErrorEncryptedFSError);
+              }
+            });
+            // Should have at least 1 success
+            expect(results.some((item) => item.status === 'fulfilled')).toBe(
+              true,
+            );
+          })
+          .afterEach(async () => {
+            // Cleaning up
+            await efs.rmdir('dir', { recursive: true });
+          }),
+        { numRuns: 50 },
+      );
+    });
```

Comment:
Should always apply the `globalThis.defaultTimeout` here. The jest library version of fast-check doesn't have this by default. On the CI/CD machines they may run this very slowly.

We should apply a "max timeout" to avoid running past the time limit on the CI/CD machines.

You could do something lower than `defaultTimeout`... like `globalThis.defaultTimeout - 1000`.

Reply:
We don't actually want a fixed number of runs. It should be based on time.

Reply:
The fixed number of runs can be a "maximum" variants we expect. But the timeout should be hard limit.

Reply:
I've added timeouts and adjusted the `numRuns` for the `fast-check` tests.